packages/store-sync/src/common.ts

```diff
@@ -115,6 +115,10 @@ export type SyncOptions = {
     blockNumber: bigint;
     logs: readonly StorageAdapterLog[];
   };
+  /**
+   * Optional webSocket URL with `wiresaw_getLogs` and `wiresaw_watchLogs` RPC support for syncing with pending logs
+   */
+  pendingLogsWebSocketRpcUrl?: string;
```

Comment:
Went back and forth on this. Originally i was just grabbing the websocket url from the public client's `chain` config. Then E2E tests were failing because anvil was configured by default _with_ a websocket URL, but it wasn't the websocket URL used in tests. That made me realise that defaulting to the chain config is probably not a good idea because often a custom RPC URL is passed to the client transport. Unfortunately there isn't a good way to grab the WebSocket URL or underlying socket from a client's transport (especially with things like `fallback` transport), so I think making the pending logs websocket URL explicit is the best option here.

Alternatively we could look for it on the chain config under `rpcUrls.wiresaw` or similar, but i'm not sure if that's better than making it an explicit option on the sync stack.

Reply:
could also rename this property to `experimentalPendingLogsWebSocketRpcUrl` to make it obvious that this is experimental and the prop might be moved in the future.